File: schedules/app/models/section.rb
# Contains logic belonging to the +Section+ model.
#
# TODO: Add more docs
class Section < ApplicationRecord
  # Each +Section+ belongs to a +Course+.
  belongs_to :course

  # Ensure all necessary fields are present.
  validates :name, presence: true
  validates :crn, presence: true
  # Unsure if necessary
  # validates :section_type, presence: true
  validates :title, presence: true
  validates :start_date, presence: true
  validates :end_date, presence: true
  validates :days, presence: true
  validates :course_id, presence: true
end

File: schedules_web/src/util/section.ts
/**
 * util/section.ts
 *
 * Common object interface for all "Section"s.
 */
export interface Section {
    id: number;
    name: string;
    title: string;
    crn: string;
    instructor: string;
    location: string;
    days: string;
    startTime: string;
    endTime: string;
}
